src/main/java/service/impl/OperationStrategyImpl.java

```diff
@@ -0,0 +1,17 @@
+package service.impl;
+
+import model.Operation;
+import service.OperationHandler;
+import service.OperationStrategy;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    @Override
+    public OperationHandler getOperationHandler(Operation operation) {
+        return switch (operation) {
+            case BALANCE -> new BalanceOperationHandler();
+            case RETURN -> new ReturnOperationHandler();
+            case SUPPLY -> new SupplyOperationHandler();
+            case PURCHASE -> new PurchaseOperationHandler();

```

Comment:
generally using switch-case is a bad practice, will be much better to replace it with `Map<Operation, OperationHandler> handlers` and pass it through constructor to here
```suggestion
        return handlers.get(operation);
```